### DISCUSSION TITLE
No more downloads of pre-built APKs

### DISCUSSION THREAD

**Discussion by khskekec** [DiaKEM/dexcom-g7-apk-patcher#6] [Announcements]
Hi,

Unfortunately i  will not offer patched APKs anymore.

The main reason for this decision are legal regulations in Germany.

On the other hand it is possible to offer software which applies patches to the original program - like we do here. 

To compensate this i plan to write a detailed tutorial to describe the patching procedure so that non-techies are also able to create a patched APK for Unix. 

Might be that there are some volunteers to do the same for Linux and Windows. 

Best regards

Selçuk

**Comment by CaptainBalou** [comment]
Nothing is more ungrateful as writing documentation - but it needs to be done to let others participate in the project and to raise overall quality.
How is it possibe to help here? Is normally use macOS to build AndroidAPS and I would normally do that for this project too. I never helped updating documentation in an community project or so. So I have no idea how I could participate here. Can you give a step-by-step tutorial how this is done?

**Reply by Atlas-Night-Out** [reply]
I can add to my own Github but no clue how to add it to this one or any other I have my own ways to do things bit I can add it to mine if you like and add a link to it.

**Comment by khskekec** [comment]
Hey @CaptainBalou,

I think the patching procedure could become obsolete in the future because there are some fearless people out there which are sharing the apk directly:


https://github.com/authorgambel/g7/blob/main/dexcom.g7.compatibility.errorcodes.aaps.v1.3.3.3527.apk

Let assume that they continue doing that for the moment... In this case the patching procedure is totally unnecessary 🎉😍. 

What u could think if is too create a documentation (written and as video maybe) for building AAPS, setting it up, integrating the g7 app and so on. 

What do you think?
